pkg/scheduler/framework/interface.go

```diff
@@ -177,6 +177,9 @@ func (s *Status) Message() string {
 	if s == nil {
 		return ""
 	}
+	if s.err != nil && s.code != Error {
```

Comment:
It shouldn't matter what code it is

```suggestion
	if s.err != nil {
```

Reply:
you could still store the message in reasons... the main thing is that we shouldn't duplicate the string.

Reply:
I don't quiet understand, I think when the code of a status is `Error`, error message will already be in reasons. So we should check code here and only concate error message in other scenarios(code not error).

And I was wrong above, we have to store the message in reasons for `s.Reasons()`

Reply:
`s.Reasons` is only called for code!=Error

Maybe we should add a panic to make sure it's not used anywhere else.

Reply:
I now store error messages in `s.reasons` only when `code != error`.
If `code==Error `,then we have all the messages in `s.err.Error()`, in that case, we can constrcut resons when we really calling `s.Reason()` or `s.Message()`, we don't have to store the same strings twice.

Is this reasonable?